From 4107c932b0a0de93c7dfb7fc75c14125ea004d79 Mon Sep 17 00:00:00 2001
From: Shivani Agarwal <shivania2@vmware.com>
Date: Fri, 9 Sep 2022 14:20:47 +0000
Subject: [PATCH 25/35] Fix bin denials for K8's deployment with containerd

Changes are added to resolve following bin denials

avc:  denied  { watch } for pid=2071 comm="kube-controller"
path="/usr/libexec/kubernetes/kubelet-plugins/volume/exec"
dev="sda3" ino=1458924 scontext=system_u:system_r:container_t:s0:c610,c650
tcontext=system_u:object_r:bin_t:s0 tclass=dir permissive=1
---
 policy/modules/contrib/container.if | 24 ++++++++++++++++++++++++
 policy/modules/contrib/container.te |  1 +
 2 files changed, 25 insertions(+)

diff --git a/policy/modules/contrib/container.if b/policy/modules/contrib/container.if
index 1dad31e..c6db6af 100644
--- a/policy/modules/contrib/container.if
+++ b/policy/modules/contrib/container.if
@@ -204,6 +204,30 @@ interface(`container_runtime_read_kubernetes_files',`
         read_files_pattern($1, kubernetes_file_t, kubernetes_file_t)
 ')
 
+########################################
+## <summary>
+##      Allow process to watch bin files.
+## </summary>
+## <desc>
+##      <p>
+##      Allow the specified domain to watch bin files
+##      </p>
+## </desc>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`container_runtime_watch_bin_files',`
+        gen_require(`
+                type bin_t;
+        ')
+
+         allow $1 bin_t:dir watch_dir_perms;
+         allow $1 bin_t:file watch_file_perms;
+')
+
 ########################################
 ## <summary>
 ##	Manage container share files.
diff --git a/policy/modules/contrib/container.te b/policy/modules/contrib/container.te
index b2a2d6d..82c85f5 100644
--- a/policy/modules/contrib/container.te
+++ b/policy/modules/contrib/container.te
@@ -1300,3 +1300,4 @@ kernel_mounton_proc(container_engine_t)
 kernel_mounton_systemd_ProtectKernelTunables(container_engine_t)
 term_mount_pty_fs(container_engine_t)
 container_runtime_read_kubernetes_files(container_t)
+container_runtime_watch_bin_files(container_t)
-- 
2.45.0

